react/src/components/SSHKeypairGenerationModal.tsx

```diff
@@ -23,18 +28,15 @@ const SSHKeypairGenerationModal: React.FC<SSHKeypairGenerationModalProps> = ({
   const { token } = theme.useToken();
   const baiClient = useSuspendedBackendaiClient();
 
-  const { data } = useTanQuery<{
-    ssh_public_key: string;
-    ssh_private_key: string;
-  }>({
+  const { data } = useTanQuery<SSHKeypair>({
     queryKey: ['refreshSSHKeypair', baiModalProps.open],
     queryFn: () => {
       return baiModalProps.open ? baiClient.refreshSSHKeypair() : null;
     },
   });
 
   const mutationToPostSSHKeypair = useTanMutation({
-    mutationFn: () => {
+    mutationFn: (data: SSHKeypair) => {
```

Comment:
`mutationToPostSSHKeypair` is not needed in this modal. Please remove it and any related codes.

Reply:
I modified It I deleted `OnRequestRefresh` and did not include the success parameter in `OnRequestClose`, because there is no POST request, so `OnRequestClose` is always called when the confirmation dialog is closed, regardless of success.